src/mutate.ts

```diff
@@ -0,0 +1,82 @@
+import type { Client } from "openapi-fetch";
+import type { MediaType, PathsWithMethod } from "openapi-typescript-helpers";
+import { useCallback, useDebugValue } from "react";
+import { type MutatorCallback, type MutatorOptions, useSWRConfig } from "swr";
+import type { PartialDeep } from "type-fest";
+import type { TypesForGetRequest } from "./types.js";
+
+// Types are loose here to support ecosystem utilities like `_.isMatch`
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type CompareFn = (init: any, partialInit: any) => boolean;
+
+/**
+ * ```ts
+ * import { isMatch } from "lodash";
```

Comment:
Doesn't isMatch return true for partial object matches since it is just a check for object values? Then wouldn't that cause a query with one parameter being the same value, but not having the other values to show as the same key?

![image](https://github.com/user-attachments/assets/cd4f49e3-2e57-48c6-8f68-d81c46113968)

Reply:
That is definitely the intent. I leave the comparator as customizable though, so the library doesn't actually require `_.isMatch`.